sql/connect/common/src/main/scala/org/apache/spark/sql/connect/client/SparkConnectClient.scala

```diff
@@ -620,8 +620,18 @@ object SparkConnectClient {
      * Configure the builder using the env SPARK_REMOTE environment variable.
      */
     def loadFromEnvironment(): Builder = {
+      lazy val isAPIModeConnect = Option(System.getProperty("spark.api.mode"))
```

Comment:
This is needed because we directly execute `ConnectRepl` (which is same as Python or R shells).